File: sdks/typescript/test/time-values.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import {
  localRfc3339Deserialize,
  localRfc3339Serialize,
  nowLocal,
  nowMillis,
  nowSecs,
  optionLocalRfc3339Deserialize,
  optionLocalRfc3339Serialize,
  toLocal,
} from "../dist/index.js";

test("time helpers preserve instant and null serialization semantics", () => {
  const before = Date.now();
  assert.ok(nowMillis() >= before);
  assert.ok(nowSecs() > 1_700_000_000);
  const source = "2026-07-09T01:50:48.876Z";
  const local = localRfc3339Serialize(source);
  assert.match(local, /[+-]\d{2}:\d{2}$/u);
  assert.equal(localRfc3339Deserialize(local).getTime(), new Date(source).getTime());
  assert.equal(toLocal(source), local);
  assert.equal(optionLocalRfc3339Serialize(null), null);
  assert.equal(optionLocalRfc3339Deserialize(null), null);
  assert.match(nowLocal(), /[+-]\d{2}:\d{2}$/u);
  assert.throws(() => localRfc3339Deserialize("bad"), TypeError);
});

test("time helper mappings are complete", () => {
  const manifest = JSON.parse(
    readFileSync(new URL("../../../contracts/sdk/parity-manifest.json", import.meta.url), "utf8"),
  );
  const entries = manifest.entries.filter(
    (entry) => entry.canonical && entry.languages.typescript.contract_test.endsWith("/time_values"),
  );
  assert.equal(entries.length, 8);
  for (const entry of entries) assert.equal(entry.languages.typescript.status, "done", entry.path);
});

File: sdks/java/src/test/java/com/echoagent/sdk/TimeValuesTest.java
package com.echoagent.sdk;

import com.fasterxml.jackson.databind.JsonNode;
import org.junit.jupiter.api.Test;

import java.nio.file.Files;
import java.nio.file.Path;
import java.time.Instant;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

class TimeValuesTest {
    @Test
    void timeHelpersPreserveInstantAndNullSemantics() {
        assertTrue(TimeValues.nowMillis() > 1_700_000_000_000L);
        assertTrue(TimeValues.nowSecs() > 1_700_000_000L);
        Instant source = Instant.parse("2026-07-09T01:50:48.876Z");
        String local = TimeValues.localRfc3339Serialize(source);
        assertTrue(local.matches(".*[+-][0-9]{2}:[0-9]{2}$"));
        assertEquals(source, TimeValues.localRfc3339Deserialize(local));
        assertEquals(source, TimeValues.optionLocalRfc3339Deserialize(local));
        assertNull(TimeValues.optionLocalRfc3339Serialize(null));
        assertNull(TimeValues.optionLocalRfc3339Deserialize(null));
        assertNotNull(TimeValues.nowLocal());
        assertThrows(IllegalArgumentException.class, () -> TimeValues.localRfc3339Deserialize("bad"));
    }

    @Test
    void timeHelperMappingsAreComplete() throws Exception {
        JsonNode manifest = JsonSupport.MAPPER.readTree(
                Files.readString(Path.of("../../contracts/sdk/parity-manifest.json")));
        int count = 0;
        for (JsonNode entry : manifest.path("entries")) {
            if (entry.path("canonical").asBoolean()
                    && entry.path("languages").path("java").path("contract_test")
                    .asText().endsWith("/time_values")) {
                count++;
                assertEquals("done", entry.path("languages").path("java").path("status").asText());
            }
        }
        assertEquals(8, count);
    }
}
